### PR TITLE
Incorporated context and bar rendering

### PR DESCRIPTION
**Pull request by MyStarrySpace**
Added contexts to pass variables between pages easily, reducing redundant backend requests and need for cache. Also added basic rendering of bars (there are still some issues, such as:

- Assignments not being sorted for consistency 
- Corner rounding calculations not implemented yet
- Entries that span multiple years are not joined properly
- Bars are rendered solid blue, rather than green or gray w/ zebra stripes

### CODE REVIEW

File: app/components/addAssignmentModal.tsx
Comment (HankC138) [inline app/components/addAssignmentModal.tsx:1]: Do lines 49-67 need to be there for what you've added or is that just a remnant from my attempt to utilize the apollo cache? If it's just from my attempt we can remove that.

Comment (MyStarrySpace) [reply app/components/addAssignmentModal.tsx:1]: I think those are leftover from when you tried to use the Apollo cache and they can be removed

**Comment by HankC138** [comment]
I'm sure we'll need to deal with a fair amount of conflicts but that's understandable with such a large rework. Thank you for this big refactor 🙏 . Should we add in my ellipsis menu functionality before or after the change as it exists based on the previous arrangment of files? No biggie either way. 

Would you like to share your linter/prettier settings so we can can sync that up as well so we have less formatting changes in our commits?

**Comment by MyStarrySpace** [comment]
> I'm sure we'll need to deal with a fair amount of conflicts but that's understandable with such a large rework. Thank you for this big refactor 🙏 . Should we add in my ellipsis menu functionality before or after the change as it exists based on the previous arrangment of files? No biggie either way.

I suppose it makes more sense to update the bigger context-related changes first, since so much of the project structure was changed, and the ellipsis menus mainly affect the sidebar and functions passed into it, which aren't a super huge change

> Would you like to share your linter/prettier settings so we can can sync that up as well so we have less formatting changes in our commits?

I'm using Ctrl+Shift+F in vscode to auto-format. Let me look up how to export those settings

**Comment by MyStarrySpace** [comment]
Correction: Shift+Alt+F, not Ctrl-Shift

**Comment by HankC138** [comment]
Good to go then feel free to resolve any conflicts and merge this one! Thanks again for all this! 🚀

**Pull request merged**